beamline.rst

```diff
@@ -1,212 +1,379 @@
+
+        NOTINITIALIZED = 0  # Actuator has not yet been set up. value is None
+        UNUSABLE = 1        # Actuator is not functional. value is None
+        READY = 2           # Actuator is functional and ready to accept new moves.
+        MOVESTARTED = 3     # Move has started. Why do we need this?
+        MOVING = 4          # Actuator is moving and does not accept move orders.
+                            # Value is defined but unstable.
+        ONLIMIT = 5         # Value is on limit. Actuator accepts move orders (?)
+        FROZEN = 6          # Actuator is functional, but cannnot be moved.
+                            # value is defined, and may be modified by HO level.
+                            # Needed for e.g. wavelength on non-tunable beamlines,
+                            # machine_current, fill_mode.
+
+      class TwoStateValue(Enum):
+
+        # There are two states, with aliases, the ACTIVE/IN/CLOSED state
+        # and the INACTIVE/OUT/OPEN
+        # As a mnemonic, you could say that 1 is for when the object is
+        # 'doing its job' (shutter closed, beamstop and frontlight in, ...)
+        # That means that for collection you need beamstop IN, and frontlight OUT
+        #
+        # NB Do we need an (oxymoronic) third state, like UNUSABLE?
+
+        INACTIVE = 0
+        OUT = 0
+        OPEN = 0
+
+        ACTIVE = 1
+        IN = 1
+        CLOSED = 1
+
+
+Specific Actuators:
+-------------------
+
+**Type float:**
+
+    *Centring motors*
+
+        Omega, Kappa, Phi, AlignmentX,  AlignmentY,  AlignmentZ,
+        CentringX, CentringY, CentringZ,  Focus
+
+        # NB, the capitalized case is to distinguish them from the current
+        #     lower-case names ('Phi' matches 'kappa_phi', not 'phi')
+        #
+        # NB There are too many motor names here, and maybe we should shorten
+        #    the list. It should be OK, though if the same motor is pointed to
+        #    by two different names. The problems is how the names match.
+        #    If you do not have 'CentringZ', which motor should you use during
+        #    centring?
+        #    Is 'Focus' always the same as 'AlignmentX'?
+        #    Does 'CentringZ' always match the same alignment motor
+        #    or does it depend on beamline geometry?
+
+    *Detector motors*
+
+        detector_distance, two_theta, detector_horizontal, detector_vertical
+
+        # detector_distance may be the only common one,
+        # but the others are clearly defined.
+        # NB horizontal and vertical are kept separate, in case some beamlines
+        #    have one without the other.
+
+    *Beamline parameters*
+
+        resolution, energy, wavelength, transmission
+
+        # NB energy/wavelength and resolution/detector_distance are both needed.
+
+    *Light intensity*
+
+        frontlight_intensity, backlight_intensity
+
+        # NB are these continuous-value floats, or rather values_list or on/off?
+
```

Comment:
Lights usually have a in/out property as well as intensity, you can see it a bit as on, off and intensity. They are moved in and out to not interfere with the beam or other equipment.

Reply:
The On/Off is handed as a separate property, below.